#!/usr/bin/env python

PACKAGE = "cob_omni_drive_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Parameters for tuning Impedance-Controller
gen.add("spring",     double_t, 0, "Spring-constant (elasticity)",                              15.0,  0.0,  50.0)
gen.add("damp",       double_t, 0, "Damping coefficient (also prop. for Velocity Feedforward)",  2.75, 0.0,  50.0)
gen.add("virt_mass",  double_t, 0, "Virtual Mass of Spring-Damper System",                       0.1,  0.0, 200.0)
gen.add("d_phi_max",  double_t, 0, "maximum angular velocity (cut-off)",                        10.0,  0.0,  50.0)
gen.add("dd_phi_max", double_t, 0, "maximum angular acceleration (cut-off)",                    40.0,  0.0, 100.0)

# Second arg is node name it will run in (doc purposes only), third is generated filename prefix
exit(gen.generate(PACKAGE, "steer_control_configure", "SteerCtrl"))
